coverage.py

```diff
@@ -262,14 +262,88 @@ def should_update(self, coverage_record):
         # cutoff_time.
         return coverage_record.timestamp < self.cutoff_time
 
+    def finalize_batch(self):
+        """Do whatever is necessary to complete this batch before moving on to
+        the next one.
+        
+        e.g. uploading a bunch of assets to S3.
+        """
+        pass
+
+    #
+    # Subclasses must implement these virtual methods.
+    #
+
+    def items_that_need_coverage(self, identifiers):
```

Comment:
This used to be a property, and any identifier restriction was tacked on afterwards. But the CoverageProvider query is a query directly on Identifier, so it can filter Identifier.id, whereas the WorkCoverageProvider query gets to Identifier via Work and LicensePool, so it needs to filter on LicensePool.identifier_id. The simplest way to implement this difference was in the subclasses.